src/api/c++/z3++.h

```diff
@@ -3676,42 +3843,42 @@ namespace z3 {
         /**
            \brief register callbacks.
            Callbacks can only be registered with user_propagators
-           that were created using a solver. 
+           that were created using a solver.
         */
 
-        void fixed(fixed_eh_t& f) { 
+        void fixed(fixed_eh_t& f) {
             assert(s);
-            m_fixed_eh = f; 
-            Z3_solver_propagate_fixed(ctx(), *s, fixed_eh); 
+            m_fixed_eh = f;
+            Z3_solver_propagate_fixed(ctx(), *s, fixed_eh);
         }
 
-        void eq(eq_eh_t& f) { 
+        void eq(eq_eh_t& f) {
             assert(s);
-            m_eq_eh = f; 
-            Z3_solver_propagate_eq(ctx(), *s, eq_eh); 
+            m_eq_eh = f;
+            Z3_solver_propagate_eq(ctx(), *s, eq_eh);
         }
 
         /**
            \brief register a callback on final-check.
            During the final check stage, all propagations have been processed.
            This is an opportunity for the user-propagator to delay some analysis
            that could be expensive to perform incrementally. It is also an opportunity
-           for the propagator to implement branch and bound optimization. 
+           for the propagator to implement branch and bound optimization.
         */
 
-        void final(final_eh_t& f) { 
+        void final_(final_eh_t& f) {
```

Comment:
This function was named `final`, which is a C++11+ keyword. I renamed it `final_` so there are no issues.

Reply:
let's maybe call it `register_final`

Reply:
Ok